### PR TITLE
fix(core): refresh the entire list of addresses after deleting an address

### PR DESCRIPTION
**Pull request by bc-yevhenii-buliuk**
…

## What/Why?
This PR fixes the display of the existing address list after removing all visible addresses on the page.
Also this PR updates the entire list of addresses after deleting an address.

## Testing
locally

**before:**

https://github.com/user-attachments/assets/bdc81450-2b76-47a5-9ea9-ff202c2d4e43

**after:**

https://github.com/user-attachments/assets/dd9f46b3-d90d-4954-a8aa-fd3010d48ebb

### CODE REVIEW

File: core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx [hunk 1]
```diff
@@ -24,13 +26,17 @@ export async function generateMetadata() {
   };
 }
 
+export enum AddressPageLimit {
+  Default = 3,
```
Comment (bc-alexsaiannyi) [inline core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx:30]: Just curious if you discuss the qty with Turner or Eric? I believe previously we have increased it to 10🤔

File: core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx [hunk 2]
```diff
@@ -24,13 +26,17 @@ export async function generateMetadata() {
...
+}
+
 export default async function Addresses({ searchParams }: Props) {
   const { before, after } = await searchParams;
 
   const data = await getCustomerAddresses({
     ...(after && { after }),
     ...(before && { before }),
-    limit: 10,
+    limit: AddressPageLimit.Default,
```
Comment (bc-alexsaiannyi) [inline core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx:39]: Also we have a default value for `limit` argument in `getCustomerAddresses` so maybe we can little bit simplify here and instead of `enum` use its default value, wdyt?💅

File: core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx [hunk 3]
```diff
@@ -43,7 +49,13 @@ export default async function Addresses({ searchParams }: Props) {
   return (
     <>
       <TabHeading heading="addresses" />
-      <AddressBook addressesCount={addressesCount} customerAddresses={addresses} key={endCursor}>
+      <AddressBook 
+        addressesCount={addressesCount}
+        customerAddresses={addresses}
+        key={endCursor}
+        pageInfo={pageInfo}
```
Comment (bc-alexsaiannyi) [inline core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx:56]: Small suggestion here - we use key for [resetting](https://react.dev/reference/react/useState#resetting-state-with-a-key) state. So if we do component's refactoring then we can go further and optimize its props in the next way.

on deleting all addresses on a Page we check if there a Previous page and if so we move there:
```
if (addressBook.length === 0 && hasPreviousPage) {
      router.push(`/account/addresses/?before=${startCursor}`);
    }
```
So basically we just need `hasPreviousPage` & `startCursor` from `PageInfo` data. Based on this we can add them to `AddressBook` Props directly and remove `key` and `PageInfo` instead. Maybe I missed smth, wdyt?

File: core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx (hunk 1 above)
Comment (bc-yevhenii-buliuk) [reply core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx]: You are right, I changed it to 3 during work and local testing. I changed it back to 10, thanks for noticing.

File: core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx (hunk 2 above)
Comment (bc-yevhenii-buliuk) [reply core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx]: Makes sense. Done!

File: core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx (hunk 3 above)
Comment (bc-yevhenii-buliuk) [reply core/app/[locale]/(default)/account/(tabs)/addresses/page.tsx]: I optimized the props passing, but removing the `key` breaks the pagination, so I left it to reset the component state when switching between pages. Thanks for the suggestion!

**Review by bc-alexsaiannyi — APPROVED** [review]

File: core/app/[locale]/(default)/account/(tabs)/addresses/_actions/delete-address.ts [hunk 4]
```diff
@@ -53,7 +64,16 @@ export const deleteAddress = async (addressId: number): Promise<State> => {
...
+      const updatedData = await getCustomerAddresses({
+        ...(after && { after }),
+        ...(before && { before }),
+      });
+
+      return {
+        status: 'success',
+        message: t('success'),
+        addresses: updatedData?.addresses || [],
+      };
```
Comment (chanceaclark) [inline core/app/[locale]/(default)/account/(tabs)/addresses/_actions/delete-address.ts:76]: Why don't we just use Next.js [revalidation](https://nextjs.org/docs/app/api-reference/functions/revalidateTag) + reload the page?

**Review by deini — APPROVED** [review]

File: core/app/[locale]/(default)/account/(tabs)/addresses/_actions/delete-address.ts (hunk 4 above)
Comment (bc-yevhenii-buliuk) [reply core/app/[locale]/(default)/account/(tabs)/addresses/_actions/delete-address.ts]: @chanceaclark thanks for the suggestion, I simplified the logic and left `revalidatePath` as it is done in adding and editing address. Could you double check the implementation?

**Review by chanceaclark — APPROVED** [review]
